src/components/Dropdown.tsx

```diff
@@ -0,0 +1,78 @@
+import cn from 'classnames';
+import { Person } from '../types/Person';
+import { useRef, useState } from 'react';
+import '../App.scss';
+
+interface Props {
+  query: string;
+  onQueryChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  filteredPeople: Person[];
+  onSelected: (person: Person) => void;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const Dropdown: React.FC<Props> = ({
+  query,
+  onQueryChange,
+  filteredPeople,
+  onSelected,
+  setQuery,
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isInputFocused, setIsInputFocused] = useState(false);
+
+  const handleFocus = () => {
+    setIsInputFocused(true);
+  };
+
+  const handleUnfocus = () => {
+    setIsInputFocused(false);
+  };
+
+  const handleSelectAndFocus = (person: Person) => {
+    onSelected(person);
+    setQuery(person.name);
+    handleUnfocus();
+  };
+
+  return (
+    <div className="dropdown is-active">
+      <div className="dropdown-trigger">
+        <input
+          ref={inputRef}
+          onFocus={handleFocus}
+          type="search"
+          placeholder="Enter a part of the name"
+          className="input"
+          data-cy="search-input"
+          value={query}
+          onChange={onQueryChange}
+        />
+      </div>
+
+      {isInputFocused && !!filteredPeople.length && (
+        <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+          <div className="dropdown-content">
+            {filteredPeople.map(person => (
+              <div
+                onClick={() => handleSelectAndFocus(person)}
+                key={person.slug}
```

Comment:
Generating keys on render using properties that might change, like 'slug', can lead to performance issues. Consider using a stable property as the key, such as an id that doesn't change.